### PR TITLE
Add TAG leadership election governance template

### PR DESCRIPTION
**Pull request by leonardpahlke**
This PR adds a template under the `toc/tags/resources/tag-formation-template` to document the leadership election process within TAGs. This PR https://github.com/cncf/toc/pull/1086 established the `/tags/resources` structure, and this PR adds content to one of the empty stub files. More PRs will follow to add content to the other template files (like: https://github.com/cncf/toc/pull/1128 and https://github.com/cncf/toc/pull/1116).

The election process is a combined version of the [TAG ENV TL](https://github.com/cncf/tag-env-sustainability/blob/main/governance/tech-lead-proposal-process.md), [TAG ENV Chair](https://github.com/cncf/tag-env-sustainability/blob/main/governance/chair-proposal-process.md) and [TAG ENV WG Chair](https://github.com/cncf/tag-env-sustainability/blob/main/governance/working-group-chair-proposal-process.md) election process. TAG ENV originally pulled the governance files from TAG Security, see [file](https://github.com/cncf/tag-security/blob/main/governance/tech-lead-proposal-process.md).

### CODE REVIEW

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 1]
```diff
@@ -1,3 +1,82 @@
...
+
+Upon close of nominations, a decision should be made within the next month.
+
+### Submitting a nomination
+
+Nominations are to be submitted to [TAG Co-Chair's email alias](mailto:cncf-tag-chairs@lists.cncf.io) with the details: <!-- ! UPDATE EMAIL ! -->
+
+* Subject "NOMINATION $(TAG Chair, TAG TL, WG XYZ Chair): $NAME-OF-NOMINEE"
+* CC the nominee
+* Justification
```
Comment (TheFoxAtWork) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:49]:
```suggestion
* Justification
* Nominee Biography
```

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 2]
```diff
@@ -1,3 +1,82 @@
...
+
+To ensure nominations are thoughtful and supported, in addition to the above requirements, a justification must enumerate on the nominee's:
+
+* TAG Community involvement
+* Leadership and collaboration skills
+* Willingness to learn and expand existing technical knowledge
+
+## Announcing nomination results
+
+The TAG leadership will release aggregate information about the nominees at the end of the nomination process. Aggregate information will not include who nominated someone.
```
Comment (TheFoxAtWork) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:82]:
```suggestion
The TAG leadership will release aggregate information about the nominees at the end of the nomination process. Aggregate information will not include who nominated someone.

## Next Steps

If the role under nomination requires TOC approval or vote, the TAG Co-chairs are responsible for coordinating and executing that in accordance with the TOC's processes (defined previously).
```

**Review by TheFoxAtWork — CHANGES_REQUESTED** [review]
Overall looks good, made some suggestions to increase clarity.

**Comment by leonardpahlke** [comment]
ref "Implementation of an updated TAG formation process" issue: https://github.com/cncf/toc/issues/1043#event-10606256466

**Comment by leonardpahlke** [comment]
I am thinking about splitting the document into two parts.

1. election procedure for TAG chairs and TAG TLs
2. election procedure for working group and project leads.

I think the procedure for working group and project leads can be simplified.

**Comment by TheFoxAtWork** [comment]
@leonardpahlke, agreed. Since working group and project leads are a different process, than TAG Chairs and TAG TLs it makes sense to separate them.

**Review by TheFoxAtWork — APPROVED** [review]

**Comment by TheFoxAtWork** [comment]
@nikhita @linsun would you both PTAL and see if this aligns with the current direction of the leadership terms and ladder effort? if it does would you provide your review so this may be merged in?

**Comment by linsun** [comment]
Sorry about the delay with lots of travel, I will review this this week and make sure the doc content is also merged in: https://docs.google.com/document/d/1xP1SXy4LW9yxI_1cs3GYQyv6wb3-JYBdXE8MYO0RnqQ/edit?usp=sharing

**Comment by linsun** [comment]
Hi @leonardpahlke this looks great. Based on our prior discussion, do you think we can add the following as well?

**Leadership Terms:**
Each leadership role is on a 2 year term once appointed.

**Elections:**
Elections should be conducted by CNCF staff. TAG leadership Election should start 1-2 months after the TOC election finishes each year, pending CNCF staff’s availability to run the elections. 

For the immediate election in 2025, roughly 50% of the roles should be made available for election where existing leaders can choose to rerun, starting with the longest serving leaders. The rest of the leads who are not up for election should plan to be up for elections in 2026.

Co-chairs can be nominated by any existing co-chairs or TOC member. TL roles should be nominated by any existing co-chairs or TLs. WG leads can be nominated by any TAG leaders.TAGs may define their own process by which they field nominations for these roles, provided they adhere to CNCF and TOC expectations of openness, transparency, and accountability.

Co-chair elections should be voted on by the TOC. Other TAG leads elections should be voted by TAG co-chairs, TLs, WG leads or other active members of the TAG.

**Vacancies**
In the event that a TAG lead vacates their seat during their term, a by-election shall be conducted to fill the position for the remainder of the term in accordance with the regular election procedure.
The TAG co-chairs may initiate a vote of no confidence in a TAG leader when the TAG leader has not performed any TAG leader tasks for 6 months. The leader shall be removed if the motion is approved by at least sixty percent (60%) of the TAG leaders along with the approval support from all of the TOC Liaisons for the TAG.

**Liaisons**
Following the TOC liaison for TAG model, we recommend that at least one TAG co-chair or/and TL serve as liaison for each TAG WG so the co-chair or/and TL can help resolve any concerns from the WG within the TAG’s own governance. This ensures TAG Leadership remains aware and conscious of the WG’s activities, deliverables, and overall health.

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 3]
```diff
@@ -1,3 +1,87 @@
...
+* Nominee Biography
+
+## Nomination requirements
+
+The `<TAG NAME>` recognizes and values the achievements of individuals who have proven themselves in this community.
+If a given nominee does not meet all the requirements, the TAG leadership will still review the nomination and justification to determine if enough intent and commitment is present to move forward with an endorsement.
+
+**Nominees must**:
+
+* Be the author of at least one pull request against the `<TAG NAME>` repository
```
Comment (JustinCappos) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:59]: A quick question:  What is the purpose of calling this out versus making it part of the list below?   It seems to already be somewhat related to the commenting on PRs, etc. issue below.

Is the author of a PR the person who created it?   Is any person who did a lot of the work on it an author?   What about someone who reviewed and commented heavily on it?   Do all PRs regardless of length / purpose / complexity count the same here?   

I know this is all subjective and I'm not proposing to define all the corner cases.   I'd actually prefer to make this less specific and less emphasized.   My concern is that I don't want someone with imposter syndrome who has been a great contributor, but who didn't create a PR to feel like their application shouldn't be put forth.

**Comment by leonardpahlke** [comment]
Thanks for the input. Will make updates to the PR this week.

**Comment by TheFoxAtWork** [comment]
@leonardpahlke any update on this?

**Comment by leonardpahlke** [comment]
Updated the PR. @linsun please take a look. 

> Liaisons
> Following the TOC liaison for TAG model, we recommend that at least one TAG co-chair or/and TL serve as liaison for each TAG WG so the co-chair or/and TL can help resolve any concerns from the WG within the TAG’s own governance. This ensures TAG Leadership remains aware and conscious of the WG’s activities, deliverables, and overall health.

I am wondering if this would be better placed in the roles document, since It's not really about the election, right?

**Comment by linsun** [comment]
Thanks @leonardpahlke! will review it shortly!

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 4]
```diff
@@ -1,3 +1,96 @@
...
+
+Leadership roles are open to all interested participants within the community. Depending on the needs of the community, as determined by the current TAG leadership team (Chairs and TLs), nominations may come from working group co-chairs, TAG co-chairs and/or tech leads and/or other community members. Each leadership role is on a 2 year term once appointed.
+
+Elections should be conducted by CNCF staff. TAG leadership Election should start 1-2 months after the TOC election finishes each year, pending CNCF staff’s availability to run the elections.
+
+For the immediate election in 2025, roughly 50% of the roles should be made available for election where existing leaders can choose to rerun, starting with the longest serving leaders. The rest of the leads who are not up for election should plan to be up for elections in 2026.
+
+co-chairs can be nominated by any existing co-chairs or TOC member. TL roles should be nominated by any existing co-chairs or TLs. WG leads can be nominated by any TAG leaders. TAGs may define their own process by which they field nominations for these roles, provided they adhere to CNCF and TOC expectations of openness, transparency, and accountability.
```
Comment (linsun) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:20]:
```suggestion
Co-chairs can be nominated by any existing co-chairs or TOC member. TL roles should be nominated by any existing co-chairs or TLs. WG leads can be nominated by any TAG leaders. TAGs may define their own process by which they field nominations for these roles, provided they adhere to CNCF and TOC expectations of openness, transparency, and accountability.
```

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 5]
```diff
@@ -1,3 +1,96 @@
...
+* Existing body of work
+* Justification provided during nomination
+
+In case there are no individuals expressing interest in taking up leadership positions, an evaluation process for the TAG / working group or project archival may be initiated, as [outlined by the TOC](https://github.com/cncf/toc/blob/main/tags/cncf-tags.md#retirement).
+
+### Leadership election timeline
+
+* **TAG Chair**:
+  * Start: CNCF staff initiates election 1-2 months after the TOC election.
+  * ominations: Open for 1-2 months. Co-chairs or TOC members can nominate.
```
Comment (linsun) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:39]:
```suggestion
  * nominations: Open for 1-2 months. Co-chairs or TOC members can nominate.
```

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 6]
```diff
@@ -1,3 +1,96 @@
...
+* Existing body of work
+* Justification provided during nomination
+
+In case there are no individuals expressing interest in taking up leadership positions, an evaluation process for the TAG / working group or project archival may be initiated, as [outlined by the TOC](https://github.com/cncf/toc/blob/main/tags/cncf-tags.md#retirement).
+
+### Leadership election timeline
+
+* **TAG Chair**:
+  * Start: CNCF staff initiates election 1-2 months after the TOC election.
+  * nominations: Open for 1-2 months. Co-chairs or TOC members can nominate.
```
Comment (linsun) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:39]:
```suggestion
  * Nominations: Open for 1-2 months. Co-chairs or TOC members can nominate.
```

File: tags/resources/tag-formation-templates/template-leadership-election-process.md [hunk 7]
```diff
@@ -1,3 +1,96 @@
...
+  * Active - Attending at least 1 `<TAG NAME>` Regular Meeting a month
+  * Engaging in community chat via Slack or lists
+  * Commenting on PRs and issues to drive suggestions to proposals, define scope, resolve clarity issues, etc.
+* Agree to the nomination
+
+*additional requirements may be defined in the specific [roles](template-roles.md) governance document* <!-- ! UPDATE THE LINK ! -->
+
+**Nominations must**:
+
+* Come from someone other than the nominee
```
Comment (linsun) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:87]: I'd in favor of self nominate if co-chair wants to continue to serve.

Comment (linsun) [inline tags/resources/tag-formation-templates/template-leadership-election-process.md:87]:
```suggestion
```

**Review by linsun — APPROVED** [review]
LGTM

corrected a few typos and removed one of the nomination requirements to allow self nomination which we allow for TOC candidates as well.

**Pull request merged**